include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,378 @@
+                            uint64_t pipeline_flags_count,
+                            const char *pipeline_flags[],
+                            const char *execution_directory);
+
+/**
+ * returns the number of containers registered in this pipeline
+ */
+uint64_t rp_runner_containers_count(rp_runner *runner);
+
+/**
+ * returns the container at the provided index
+ *
+ * runner cannot be null, index must be less than
+ * rp_runner_containers_count(runner)
+ */
+rp_container *rp_runner_get_container(rp_runner *runner, uint64_t index);
+
+/**
+ * returns the name of the nth container registered inside the runner
+ */
+const char *rp_container_get_name(rp_container *container);
+
+/**
+ *
+ * exactly like rp_runner_create except that instead of the path to a
+ * pipeline yaml file this overload accepts the string containing the pipeline
+ * directly
+ *
+ */
+rp_runner *rp_runner_create_from_string(uint64_t pipelines_count,
+                                        const char *pipelines[],
+                                        uint64_t pipeline_flags_count,
+                                        const char *pipeline_flags[],
+                                        const char *execution_directory);
+
+/**
+ * Exactly like rp_runner_create, except that it is run without loading and
+ * saving the state of the pipeline before and after the execution, and that no
+ * flags are provided.
+ */
+rp_runner *rp_runner_create_memory_only(const char *pipeline_path,
+                                        uint64_t flags_count,
+                                        const char *flags[]);
+
+/**
+ *  Triggers the serialization of the pipeline on disk.
+ *  Returns 0 if a error happened, 1 otherwise.
+ *
+ *  runner cannot be null
+ */
+bool rp_runner_store_containers(rp_runner *runner);
+
+/**
+ * deletes the runner object and destroys all the resourced acquired by it.
+ */
+void rp_runner_destroy(rp_runner *runner);
+
+/**
+ * returns the amount of steps present in the runner.
+ *
+ * runner cannot be null
+ */
+uint64_t rp_runner_steps_count(rp_runner *runner);
+
+const inline uint64_t NO_KNOWN_STEP = UINT64_MAX;
+
+/**
+ * returns the index of the step with the provided name.
+ * returns NO_KNOWN_STEP if no step with such name existed.
+ *
+ * runner cannot be null
+ */
+uint64_t rp_runner_step_name_to_index(rp_runner *runner, const char *name);
+
+/**
+ * returns the step with the provided index, or null if not such step existed.
+ *
+ * runner cannot be null
+ */
+rp_step *rp_runner_get_step(rp_runner *runner, uint64_t index);
+rp_step *rp_runner_get_step_from_name(rp_runner *runner, const char *name);
+
+/**
+ * returns the step name
+ *
+ * step cannot be null
+ */
+const char *rp_step_get_name(rp_step *step);
+
+/**
+ *
+ * returns the container checkpoint associated to the provided container,
+ *
+ * step cannot be null
+ *
+ */
+rp_container_checkpoint *
+rp_step_get_checkpoint(rp_step *step, rp_container *container);
```

Comment:
Mmmmh, `checkpoint`. We never speak about checkpoints anywhere else.
I know I might have been pushing from this, but at this stage I think we'd like someone reading the `revng-pipeline` doc not to be puzzled by the C API.
Let's reconsider `checkpoint` for something closer to what the rest of the world calls them.